src/x/server/tls_mode.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// Package server implements a network server.
+package server
+
+import "fmt"
+
+const (
+	// TLSDisabled allows plaintext connections only
+	TLSDisabled TLSMode = iota
+	// TLSPermissive allows both TLS and plaintext connections
+	TLSPermissive
+	// TLSEnforced allows TLS connections only
+	TLSEnforced
+)
+
+// TLSMode represents the TLS mode
+type TLSMode uint16
+
+func (t *TLSMode) UnmarshalText(mode []byte) error {
```

Comment:
nit: in statsdex we use [enumer](https://github.com/dmarkham/enumer) to deal with this boilerplate, check any `generate.go` file, i think this will also allow you to use it directly in config structures instead of a string with regexes